### ISSUE TITLE
Does this work with vipertraq

### ISSUE THREAD

**Issue by Hucksleyy91** [karlg100/homebridge-smartstart#1]
Hey does this work with the new vipertraq accounts in Australia :)

**Comment by Hucksleyy91** [comment]
It’s through directed in Australia

**Comment by karlg100** [comment]
No idea...  but give it a shot!  You may have to change the host, if it’s done thru another Australian directed website.  Try logging into directed directly and compare the hostnames.

**Comment by Hucksleyy91** [comment]
I paid for sub subscription I can use the app but not this plugin 
How do I change the host :)

**Comment by Hucksleyy91** [comment]
https://colt.calamp-ts.com
Is this the current one ?
I can log in here with my old account but not with my new account after updating from the vsq250 to the 500 because they turned off 2g, :D

**Comment by karlg100** [comment]
it sounds like they've changed backend providers.

 I don't have a good answer as I don't have that model and don't have the new service to test and engineer a new plugin.

**Comment by claywd** [comment]
@karlg100 I don't have a lot of experience around the old server but I captured the network activity off the ios app and the new server is 

remote control url: `https://www.vcp.cloud/v1/auth/login` requires username and password (in plain text) returns bearer token

`/v1/devices/command `
`/v1/devices/assets/history`
`/v1/devices/search/null?`

Login url: `services.mysmartstart.com/api/AuthenticationService/UserLogin` requires app token and returns session tokens

I've got the basic functionality mapped out and it seems consistent with the current library (aside from the host and possibly context paths). If you have time to adjust the library and need more details I can share but I need to scrub/cleanup the data and feeling a smidge ~lazy~tired atm.

**Comment by Hucksleyy91** [comment]
Nice so we know there new portal;) but I can’t log in like I could on the old portal :(

**Comment by claywd** [comment]
yeah I started looking at adapting the old library and basically everything evolved.. albeit about 5 years late but it has evolved. Heres how to login 

```curl -H 'Host: www.vcp.cloud' -H 'Accept: application/json' -H 'User-Agent: SmartStart/342 CFNetwork/978.0.7 Darwin/18.7.0' -H 'Accept-Language: en-us' --data "username=<USERNAME_OR_EMAIL>&password=<PASSWORD>" --compressed 'https://www.vcp.cloud/v1/auth/login'```

Fairly certain they are running two apis in tandem though and they are looking to switch to a new version as seen here. 

```curl -H 'Host: services.mysmartstart.com' -H 'x-dei-app-token: <PULL FROM IOS TRAFFIC>' -H 'Accept: application/json' -H 'Accept-Language: en-us' -H 'User-Agent: SmartStart/342 CFNetwork/978.0.7 Darwin/18.7.0' -H 'x-dei-handset-id: <PULL FROM IOS TRAFFIC' -H 'x-dei-app-version: 5.1' -H 'x-dei-app-platform: iPhone OS' -H 'x-dei-app-brand: Viper' --data "username=<USERNAME_OR_EMAIL>&password=<PASSWORD>" --compressed 'https://services.mysmartstart.com/api/AuthenticationService/UserLogin'``` 

Current system uses username/password but I think their next system is going to switch to tokens completely... which I suppose kind of sucks... since anyone who uses the homebridge plugin at that point will need their traffic sniffed for tokens but then again it might be the last step before they officially support the AI assistants. Maybe they'll let us register devices and retrieve tokens via the portal. 

Anyways, I'm going to start working on a PR for the library but the whole plugin might have to be updated as well in the near future. 

Also, if you aren't familiar with setting tokens in headers for request then you might not want to proceed further. There are usually security tools that scan for this sort of query, where the headers are malformed and unauthorized attempts were made against the api, in an attempt to predict hacker traffic and they will just work to make things harder.

**Comment by claywd** [comment]
Update. did a fair bit of work on a couple of other projects and I've been super tied up. I'm looking to make this happen over the next few weeks though. :)

I'll be publishing a new library.

**Comment by Hucksleyy91** [comment]
I’ll be keen to test it out 
I’ve been messaging the app developer for Vipertraq asking when they will update or allow me to update the app to include Siri Shortcuts,

**Comment by Hucksleyy91** [comment]
Any updates?

**Comment by Hucksleyy91** [comment]
> Update. did a fair bit of work on a couple of other projects and I've been super tied up. I'm looking to make this happen over the next few weeks though. :)
> 
> I'll be publishing a new library.

Hey mate wondering if you got any closer with this :)
